aten/src/ATen/native/UnaryOps.cpp

```diff
@@ -66,14 +66,43 @@ Tensor& asin_out(Tensor& result, const Tensor& self) { return unary_op_impl_out(
 Tensor asin(const Tensor& self) { return unary_op_impl(self, at::asin_out); }
 Tensor& asin_(Tensor& self) { return unary_op_impl_(self, at::asin_out); }
 
-Tensor& abs_out(Tensor& result, const Tensor& self) { return unary_op_impl_out(result, self, abs_stub); }
+Tensor& abs_out(Tensor& result, const Tensor& self) {
+  // Mimics abs returning a floating point tensor for complex inputs by default
+  // Note: This is done by running the operation as usual, then checking
+  // for a valid cast from float, not complex, and copying the operation's
+  // result to the expected result type.
+  if (self.is_complex() && !result.is_complex()) {
+    // Runs the function complex->complex, as TensorIterator expects
+    Tensor complex_result = at::empty({0}, self.options());
+    auto iter = TensorIterator::unary_op(complex_result, self,
+      /*check_mem_overlap=*/false);
+    abs_stub(iter.device_type(), iter);
+
+    const auto float_type = c10::toValueType(c10::typeMetaToScalarType(self.dtype()));
+    TORCH_CHECK(canCast(float_type, c10::typeMetaToScalarType(result.dtype())),
+          "result type ", float_type, " can't be cast to the desired output type ",
+          result.dtype());
+
+    // Copies the complex result to the actual result, resizing if needed
+    if (!result.sizes().equals(complex_result.sizes())) {
```

Comment:
might be simpler to read if we do:
```
if (result.sizes() != self.sizes()) {
    result.resize_as_(self);
  }
```

Reply:
or just get rid of the check, which will be slower but this is already going to be really slow.